api/client/ffs.go

```diff
@@ -0,0 +1,244 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	cid "github.com/ipfs/go-cid"
+	ff "github.com/textileio/powergate/ffs"
+	"github.com/textileio/powergate/ffs/api"
+	pb "github.com/textileio/powergate/ffs/pb"
+)
+
+type ffs struct {
+	client pb.APIClient
+}
+
+// JobEvent represents an event for Watching a job
+type JobEvent struct {
+	Job ff.Job
+	Err error
+}
+
+func (f *ffs) Create(ctx context.Context) (string, string, error) {
+	r, err := f.client.Create(ctx, &pb.CreateRequest{})
+	if err != nil {
+		return "", "", err
+	}
+	return r.ID, r.Token, nil
+}
+
+func (f *ffs) ID(ctx context.Context) (ff.ApiID, error) {
+	resp, err := f.client.ID(ctx, &pb.IDRequest{})
+	if err != nil {
+		return ff.EmptyInstanceID, err
+	}
+	return ff.ApiID(resp.ID), nil
+}
+
+func (f *ffs) WalletAddr(ctx context.Context) (string, error) {
+	resp, err := f.client.WalletAddr(ctx, &pb.WalletAddrRequest{})
+	if err != nil {
+		return "", err
+	}
+	return resp.Addr, nil
+}
+
+func (f *ffs) GetDefaultCidConfig(ctx context.Context, c cid.Cid) (*pb.GetDefaultCidConfigReply, error) {
+	return f.client.GetDefaultCidConfig(ctx, &pb.GetDefaultCidConfigRequest{Cid: c.String()})
+}
+
+func (f *ffs) GetCidConfig(ctx context.Context, c cid.Cid) (*pb.GetCidConfigReply, error) {
+	return f.client.GetCidConfig(ctx, &pb.GetCidConfigRequest{Cid: c.String()})
+}
+
+func (f *ffs) SetDefaultCidConfig(ctx context.Context, config ff.DefaultCidConfig) error {
+	req := &pb.SetDefaultCidConfigRequest{
+		Config: &pb.DefaultCidConfig{
+			Hot: &pb.HotConfig{
+				Enabled:       config.Hot.Enabled,
+				AllowUnfreeze: config.Hot.AllowUnfreeze,
+				Ipfs: &pb.IpfsConfig{
+					AddTimeout: int64(config.Hot.Ipfs.AddTimeout),
+				},
+			},
+			Cold: &pb.ColdConfig{
+				Enabled: config.Cold.Enabled,
+				Filecoin: &pb.FilConfig{
+					RepFactor:      int64(config.Cold.Filecoin.RepFactor),
+					DealDuration:   int64(config.Cold.Filecoin.DealDuration),
+					ExcludedMiners: config.Cold.Filecoin.ExcludedMiners,
+					CountryCodes:   config.Cold.Filecoin.CountryCodes,
+					Renew: &pb.FilRenew{
+						Enabled:   config.Cold.Filecoin.Renew.Enabled,
+						Threshold: int64(config.Cold.Filecoin.Renew.Threshold),
+					},
+				},
+			},
+		},
+	}
+	_, err := f.client.SetDefaultCidConfig(ctx, req)
+	return err
+}
+
+func (f *ffs) Show(ctx context.Context, c cid.Cid) (*pb.ShowReply, error) {
+	return f.client.Show(ctx, &pb.ShowRequest{
+		Cid: c.String(),
+	})
+}
+
+func (f *ffs) Info(ctx context.Context) (*pb.InfoReply, error) {
+	return f.client.Info(ctx, &pb.InfoRequest{})
+}
+
+func (f *ffs) Watch(ctx context.Context, jids ...ff.JobID) (<-chan JobEvent, error) {
```

Comment:
I wonder if adding some comments to client APIs would be useful. In particular in this kind of APIs where proper canceling of the `ctx` is needed to free up resources.

APIs that aren't stream-based, canceling the `ctx` isn't necessary since the request will finish and things will be freed up on the server-side.

Not really a particular comment of this method, nor this repo... but in general on stream-based gRPC client methods.

Reply:
Yea good call. Will make an issue for it.